lib/ace/snippets.js

```diff
@@ -943,6 +941,9 @@ var TabstopManager = function(editor) {
                 this.$openTabstops[index] = dest;
             }
             this.addTabstopMarkers(dest);
+            dest.rangeList = dest.rangeList || new RangeList();
+            dest.rangeList.$bias = 0;
+            dest.rangeList.addList(dest);
```

Comment:
I guess at this point you could also clear the array in `dest` right, since you've already added it to its `rangeList` ?

Reply:
Could we add this logic before `addTabstopMarkers` is called?